src/test/groovy/com/solidbrain/ContactAndNewDealSpec.groovy

```
@@ -103,78 +92,67 @@ class ContactsTest extends Specification {
 
     def "should create deal if the newly created contact is a company and the owner of the newly created contact is a sales representative"() {
         given:
-        sampleCompanyName
-        def isContactACompany = true
         sampleSalesRepresentativeId
 
         when:
+        def isContactACompany = true
         Contact sampleContact = baseClient.contacts().create([name : sampleCompanyName,
                                                               is_organization:  isContactACompany,
                                                               owner_id: sampleSalesRepresentativeId])
-        Deal sampleDeal = null
-        await().atMost(30, SECONDS).until {
-            sampleDeal = baseClient.deals().list([contact_id: sampleContact.id])[0]
-        }
 
         then:
-        sampleDeal
+        def newDealFound = true
+        try {
+            await().atMost(30, SECONDS).pollInterval(1, SECONDS).ignoreExceptions().until {
```

let the exception be thrown, it's enough information